transport/controller.cc

```diff
@@ -332,14 +332,22 @@ future<> controller::subscribe_server(sharded<cql_server>& server) {
     return server.invoke_on_all([this] (cql_server& server) {
         _mnotifier.local().register_listener(server.get_migration_listener());
         _lifecycle_notifier.local().register_subscriber(server.get_lifecycle_listener());
+        if (!_used_by_maintenance_socket) {
```

Comment:
> Make cql server (but not maintenance server) is subscribed to qos
> configuration change.

Why is it necessary not to subscribe to the events in maintenance mode? I feel this should be explained in the commit message.